File: eco-core/core-plugin/src/main/java/com/willfp/reforges/effects/effects/EffectRewardKill.java
package com.willfp.reforges.effects.effects;

import com.willfp.eco.core.config.interfaces.JSONConfig;
import com.willfp.eco.core.events.EntityDeathByEntityEvent;
import com.willfp.reforges.effects.Effect;
import com.willfp.reforges.vault.EconomyHandler;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class EffectRewardKill extends Effect {
    /**
     * Create a new effect.
     */
    public EffectRewardKill() {
        super("reward_kill");
    }

    @Override
    public void onKill(@NotNull final LivingEntity killer,
                       @NotNull final LivingEntity victim,
                       @NotNull final EntityDeathByEntityEvent event,
                       @NotNull final JSONConfig config) {
        if (!(killer instanceof Player player)) {
            return;
        }

        EconomyHandler.getInstance().depositPlayer(player, config.getDouble("amount"));
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/reforges/effects/effects/EffectRewardKill.kt
package com.willfp.reforges.effects.effects

import com.willfp.eco.core.config.interfaces.JSONConfig
import com.willfp.eco.core.events.EntityDeathByEntityEvent
import com.willfp.eco.core.integrations.economy.EconomyManager
import com.willfp.reforges.effects.Effect
import org.bukkit.entity.LivingEntity
import org.bukkit.entity.Player

class EffectRewardKill : Effect("reward_kill") {
    override fun onKill(killer: Player,
                        victim: LivingEntity,
                        event: EntityDeathByEntityEvent,
                        config: JSONConfig) {
        EconomyManager.giveMoney(killer, config.getDouble("amount"))
    }
}